test/renamer_test.dart

```diff
@@ -0,0 +1,174 @@
+      test(r'to entire match', () {
+        var renamer = Renamer.simple(r'.+ to prefix-\0');
+        expect(renamer.rename('a'), equals('prefix-a'));
+        expect(renamer.rename('abc'), equals('prefix-abc'));
+      });
+
+      test(r'to group', () {
+        var renamer = Renamer.simple(r'/(.+)-suffix/\1/');
+        expect(renamer.rename('a-suffix'), equals('a'));
+        expect(renamer.rename('abc-suffix'), equals('abc'));
+        expect(renamer.rename('abc'), isNull);
+        expect(renamer.rename('-suffix'), isNull);
+      });
+    });
+
+    group('escaping', () {
+      test(r'spaces', () {
+        var renamer = Renamer.simple(r'a\ b to x\ y');
+        expect(renamer.rename('a b'), equals('x y'));
+      });
+
+      test(r'normal delimiters', () {
+        var renamer = Renamer.simple(r'/a\/b/x\/y/');
+        expect(renamer.rename('a/b'), equals('x/y'));
+      });
+
+      test(r'semicolons', () {
+        var renamer = Renamer.simple(r'a\;b to x\;y');
+        expect(renamer.rename('a;b'), equals('x;y'));
+      });
+    });
+
+    group('multiple keys', () {
+      test('sed-style syntax', () {
+        var renamer =
+            Renamer.map(r'url=.*/(\w+)/lib/mixins=\1=', ['namespace', 'url']);
+        expect(
+            renamer.rename(
+                {'namespace': 'mixins', 'url': 'path/button/lib/mixins'}),
+            equals('button'));
+      });
+
+      test('space syntax', () {
+        var renamer =
+            Renamer.map(r'url .*/(\w+)/lib/mixins to \1', ['namespace', 'url']);
+        expect(
+            renamer.rename(
+                {'namespace': 'mixins', 'url': 'path/button/lib/mixins'}),
+            equals('button'));
+      });
+    });
+  });
+
+  group('multiple statements', () {
+    test('separated by semicolon', () {
+      var renamer = Renamer.simple('/a/b/;/x/y/');
+      expect(renamer.rename('a'), equals('b'));
+      expect(renamer.rename('b'), isNull);
+      expect(renamer.rename('x'), equals('y'));
+      expect(renamer.rename('y'), isNull);
+    });
+
+    test('separated by line break', () {
+      var renamer = Renamer.simple('/a/b/\n/x/y/');
+      expect(renamer.rename('a'), equals('b'));
+      expect(renamer.rename('b'), isNull);
+      expect(renamer.rename('x'), equals('y'));
+      expect(renamer.rename('y'), isNull);
+    });
+
+    test('separated by semicolon and line break', () {
+      var renamer = Renamer.simple('/a/b/;\n/x/y/;');
+      expect(renamer.rename('a'), equals('b'));
+      expect(renamer.rename('b'), isNull);
+      expect(renamer.rename('x'), equals('y'));
+      expect(renamer.rename('y'), isNull);
+    });
+
+    test('only first matching statement is applied', () {
+      var renamer = Renamer.simple('/.*/all/;/old/wrong/;/all/wrong/');
+      expect(renamer.rename('old'), equals('all'));
+      expect(renamer.rename('wrong'), equals('all'));
+      expect(renamer.rename('all'), equals('all'));
+    });
+  });
+
+  group('invalid syntax', () {
```

Comment:
Also:

* Empty string
* Three clauses but the middle one isn't `to`
* Four clauses but the first one isn't a known key

Reply:
Done. I decided to make a renamer with no statements not actually an error (it just will always return null), so that if someone wants to run this migrator just for `--unreferencable=all` it should still work.